tests/unit/resource/suspend_runtime.cpp

```diff
@@ -0,0 +1,59 @@
+//  Copyright (c) 2017 Mikael Simberg
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// Simple test verifying basic resource_partitioner functionality.
+
+#include <hpx/hpx_start.hpp>
+#include <hpx/hpx_suspend.hpp>
+#include <hpx/include/async.hpp>
+#include <hpx/include/threadmanager.hpp>
+#include <hpx/include/threads.hpp>
+#include <hpx/util/detail/yield_k.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+int main(int argc, char* argv[])
+{
+    std::vector<std::string> cfg =
+    {
+        "hpx.os_threads=4"
+    };
+
+    hpx::start(nullptr, argc, argv, cfg);
+
+    // Wait for runtime to start
+    hpx::runtime* rt = hpx::get_runtime_ptr();
+    hpx::util::detail::yield_while([rt]()
+        {
+            return rt->get_state() < hpx::state_running;
+        }, "");
```

Comment:
This tells me that a) `yield_while` shouldn't be in detail namespace and b) that its second argument should have a default value (or it has to be exposed through a different API).

Reply:
Fair point. Would it be okay if I move only `yield_while` into `hpx/util/yield_while.hpp` and leave `yield_k` where it is? I see `yield_k` as being a lower level helper function.

Reply:
Sounds good to me. Could be done in a separate PR, though.

Reply:
Ok, good. I'll do this in a separate PR as soon as this is merged.